dependencies.yaml

```diff
@@ -17,11 +17,11 @@
 # under the License.
 #
 
-boost : 1.83.0
-cmake: 3.24.2
+boost : 1.84.0
+cmake: 3.28.3
 protobuf: 3.20.0
-zlib: 1.2.12
-zstd: 1.5.2
-snappy: 1.1.9
-openssl: 1.1.1v
-curl: 8.4.0
+zlib: 1.3.1
+zstd: 1.5.5
+snappy: 1.1.10
```

Comment:
Snappy 1.1.0 seems not work with the latest Apple Clang

```
snappy-1.1.10/snappy.cc:1293:37: error: comparison of integers of different signs: 'unsigned long' and 'ptrdiff_t' (aka 'long') [-Werror,-Wsign-compare]
```

We might need some fix on the build script.